Document/0x06e-Testing-Cryptography.md

```diff
@@ -48,12 +95,51 @@ The following is an example of the APIs usage:
 int result = SecRandomCopyBytes(kSecRandomDefault, 16, randomBytes);
 ```
 
+Note: if other mechanims are used for random numbers in the code: verify that these are either wrappers around the APIs mentioned above review them for their secure-randomness. Often this is too hard, which means you can best stick with the implementation above.
+
+#### Dynamic Analysis
+If you want to test for randomness, you can try to capture a large set of numbers and check with the Burpsuite its [sequencer](https://portswigger.net/burp/documentation/desktop/tools/sequencer "Sequencer") to see how good the quality of the randomness is.
+
+### Testing Key Management
+
+#### Overview
+There are various methods on how to store the key on the device. Not storing a key at all will ensure that no keymaterial can be dumped. This can be archieved by using a Password Key Derivation function, such as PKBDF-2.
+<TODO: ADD PKBDF-2 EXAMPLE>
+
+When you need to store the key. Then we recommend ot use the Keychain, as long as the protection class chosen is not `kSecAttrAccessibleAlways`. Storing keys in any other location, such as the `NSUserDefaults`, Propertylists or by any other sink from Coredata or Realm, is usually not equally strong as using the KeyChain.
```

Comment:
When you need to store the key. Then we recommend ot use the Keychain -> When you need to store the key, it is recommended to use the Keychain
is usually not equally strong as using the KeyChain -> is usually less secure than using the KeyChain